addons/website_sale/models/product_product.py

```diff
@@ -75,8 +75,9 @@ def _get_images(self):
         """
         self.ensure_one()
         variant_images = list(self.product_variant_image_ids)
-        if self.image_variant_1920:
-            # if the main variant image is set, display it first
+        if self.image_variant_1920 or not self.product_tmpl_id['image_1920']:
+            # If the main variant image is set or if it is not set and the
+            # template image is also not set, then display it first.
```

Comment:
Not sure why this change? :thinking:

Reply:
This is to avoid the case of the picture. In this case, the products has multiple variants and its main image has been deleted. However, without this modification, the main image (that has been deleted) is placed at the end of all the secondary images. IMO, this is problematic for two reasons: First, this behavior is specific for products with variants. Second, the result is a last undeletable empty image (as it is the main image, it should not be possible to delete it)
![image](https://user-images.githubusercontent.com/114914971/224647987-7eb4d27e-5400-47ac-8d39-f36df1d67811.png)